src/SURFnet/SPRegistration/Grid/GridConfiguration.php

```diff
@@ -0,0 +1,217 @@
+                        return 'Draft';
+
+                    default:
+                        return 'Unknown';
+                }
+            }
+        );
+    }
+
+    /**
+     * @param Grid $grid
+     */
+    private function configureRowActions(Grid $grid)
+    {
+        $rowAction = new RowAction('view', 'admin.subscription.view');
+        $grid->addRowAction($rowAction);
+
+        $rowAction = new RowAction('edit', 'form', false, '_blank');
+        $rowAction->manipulateRender(
+            function (RowAction $action, Row $row) {
+                if ($row->getField('status') == Subscription::STATE_FINISHED) {
+                    return null;
+                }
+
+                return $action;
+            }
+        );
+        $grid->addRowAction($rowAction);
+
+        $rowAction = new RowAction('finish', 'admin.subscription.finish');
+        $rowAction->manipulateRender(
+            function (RowAction $action, Row $row) {
+                if ($row->getField('status') !== Subscription::STATE_DRAFT) {
+                    return null;
+                }
+
+                return $action;
+            }
+        );
+        $grid->addRowAction($rowAction);
+
+        $rowAction = new RowAction('draft', 'admin.subscription.draft');
+        $rowAction->manipulateRender(
+            function (RowAction $action, Row $row) {
+                if ($row->getField('status') !== Subscription::STATE_FINISHED) {
+                    return null;
+                }
+
+                return $action;
+            }
+        );
+        $grid->addRowAction($rowAction);
+
+        $rowAction = new RowAction('archive', 'admin.subscription.archive');
+        $rowAction->manipulateRender(
+            function (RowAction $action, Row $row) {
+                if ($row->getField('archived') === true) {
+                    return null;
+                }
+                if ($row->getField('status') !== Subscription::STATE_FINISHED) {
+                    return null;
+                }
+
+                return $action;
+            }
+        );
+        $grid->addRowAction($rowAction);
+
+        $this->addLinkToJanus($grid);
+    }
+
+    /**
+     * @param Grid $grid
+     */
+    private function addLinkToJanus(Grid $grid)
+    {
+        $descriptorRepository = $this->janusDescriptorRepository;
+        $rowAction = new RowAction('janus', 'admin.subscription.janus');
+        $rowAction->manipulateRender(
+            function (RowAction $action, Row $row) use ($descriptorRepository) {
+                $subscription = $row->getEntity();
+
+                if (!$subscription instanceof Subscription) {
+                    return null;
+                }
+
+                $entityId = $subscription->getEntityId();
+
+                if (empty($entityId)) {
+                    return null;
+                }
+
+                $connectionDescriptor = $descriptorRepository->findByName(
+                    $entityId
+                );
```

Comment:
Wellicht, aan de andere kant mocht een draft subscription nu onverhoopt in Janus staan dan zie je dat meteen.